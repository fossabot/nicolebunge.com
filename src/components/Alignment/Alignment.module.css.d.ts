declare const styles: {
  readonly alignment: string;
  readonly alignmentAlignCenter: string;
  readonly alignmentAlignEnd: string;
  readonly alignmentAlignStart: string;
  readonly alignmentJustifyCenter: string;
  readonly alignmentJustifyEnd: string;
  readonly alignmentJustifySpaceBetween: string;
  readonly alignmentJustifyStart: string;
};
export = styles;
