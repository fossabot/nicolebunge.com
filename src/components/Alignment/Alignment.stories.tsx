import Alignment from '.';

export default {
  component: Alignment,
};

export const Default = {
  args: {
    children: 'Alignment',
  },
};
