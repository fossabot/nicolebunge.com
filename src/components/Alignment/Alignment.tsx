import classNames from 'classnames';
import capitalize from 'lodash/capitalize';
import React, { HTMLAttributes } from 'react';
import * as styles from './Alignment.module.css';

type Align = 'center' | 'end' | 'start';
type AlignClass = 'alignmentAlignCenter' | 'alignmentAlignEnd' | 'alignmentAlignStart';
type Justify = 'center' | 'end' | 'space-between' | 'start';
type JustifyClass =
  | 'alignmentJustifyCenter'
  | 'alignmentJustifyEnd'
  | 'alignmentJustifySpaceBetween'
  | 'alignmentJustifyStart';

export interface AlignmentProps extends HTMLAttributes<HTMLDivElement> {
  align?: Align;
  justify?: Justify;
}

const Alignment: React.FC<AlignmentProps> = (props) => {
  const { align, className, justify, ...otherProps } = props;

  return (
    <div
      className={classNames(className, {
        [styles.alignment]: align || justify,
        [styles[`alignmentAlign${capitalize(align)}` as AlignClass]]: align,
        [styles[`alignmentJustify${capitalize(justify)}` as JustifyClass]]: justify,
      })}
      {...otherProps}
    />
  );
};

export default Alignment;
