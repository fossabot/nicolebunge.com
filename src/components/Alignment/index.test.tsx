/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Alignment from '.';

describe('Alignment component', () => {
  it('should render correctly', () => {
    const { container } = render(<Alignment />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render content correctly', () => {
    const { container } = render(<Alignment>Content</Alignment>);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render align `center` correctly', () => {
    const { container } = render(<Alignment align="center" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render align `end` correctly', () => {
    const { container } = render(<Alignment align="end" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render align `start` correctly', () => {
    const { container } = render(<Alignment align="start" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render justify `center` correctly', () => {
    const { container } = render(<Alignment justify="center" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render justify `end` correctly', () => {
    const { container } = render(<Alignment justify="end" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render justify `space-between` correctly', () => {
    const { container } = render(<Alignment justify="space-between" />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render justify `start` correctly', () => {
    const { container } = render(<Alignment justify="start" />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
