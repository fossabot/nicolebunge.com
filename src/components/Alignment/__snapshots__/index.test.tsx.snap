// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alignment component should render align \`center\` correctly 1`] = `
<div
  class=""
/>
`;

exports[`Alignment component should render align \`end\` correctly 1`] = `
<div
  class=""
/>
`;

exports[`Alignment component should render align \`start\` correctly 1`] = `
<div
  class=""
/>
`;

exports[`Alignment component should render content correctly 1`] = `
<div
  class=""
>
  Content
</div>
`;

exports[`Alignment component should render correctly 1`] = `
<div
  class=""
/>
`;

exports[`Alignment component should render justify \`center\` correctly 1`] = `
<div
  class="undefined"
/>
`;

exports[`Alignment component should render justify \`end\` correctly 1`] = `
<div
  class="undefined"
/>
`;

exports[`Alignment component should render justify \`space-between\` correctly 1`] = `
<div
  class="undefined"
/>
`;

exports[`Alignment component should render justify \`start\` correctly 1`] = `
<div
  class="undefined"
/>
`;
